src/Yoti.Auth/Profile.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Collections.Generic;
+
+namespace Yoti.Auth
+{
+    public class Profile
+    {
+        private Dictionary<string, BaseAttribute> _attributes;
+
+        internal Profile()
+        {
+            _attributes = new Dictionary<string, BaseAttribute>();
+        }
+
+        internal Profile(Dictionary<string, BaseAttribute> attributes)
+        {
+            _attributes = attributes;
+        }
+
+        internal void Add(BaseAttribute value)

```

Comment:
I wonder if this can still accept _YotiAttribute<T>_ as a parameter.
Then the fact that it is actually working with _BaseAttribute_ is kept entirely internal, and there is no possibility of adding the wrong kind of _BaseAttribute_ to the Dictionary.

Reply:
Yep just wrote a test for that scenario and it works, will check in